tendermint/src/lite/verifier.rs

```diff
@@ -15,30 +15,39 @@ use crate::lite::error::{Error, Kind};
 use crate::lite::{
     Commit, Header, Height, Requester, SignedHeader, TrustThreshold, TrustedState, ValidatorSet,
 };
+use anomaly::ensure;
+use std::ops::Add;
 
 /// Returns an error if the header has expired according to the given
 /// trusting_period and current time. If so, the verifier must be reset subjectively.
 fn is_within_trust_period<H>(
     last_header: &H,
-    trusting_period: &Duration,
-    now: &SystemTime,
+    trusting_period: Duration,
+    now: SystemTime,
 ) -> Result<(), Error>
 where
     H: Header,
 {
-    match now.duration_since(last_header.bft_time().into()) {
-        Ok(passed) => {
-            if passed > *trusting_period {
-                return Err(Kind::Expired {
-                    at: last_header.bft_time().into() + *trusting_period,
-                    now: *now,
-                }
-                .into());
-            }
-            Ok(())
+    let header_time: SystemTime = last_header.bft_time().into();
+    let expires_at = header_time.add(trusting_period);
+    // Ensure now > expires_at.
+    // TODO: shouldn't this be <= instead below:
```

Comment:
See discussion: https://github.com/interchainio/tendermint-rs/pull/149#discussion_r377874816

Reply:
The spec mentions `return header.Time + trustedPeriod > now` (for the non-error'ing case) and the go code also behaves accordingly. So this should indeed be `<=`!

Reply:
Changed accordingly in 301958fdbcc6763ec31390c99958a29984f81ebe